src/libraries/System.Net.NameResolution/src/System/Net/NameResolutionPal.Unix.cs

```diff
@@ -8,13 +8,17 @@
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+using System.Runtime.Versioning;
 
 namespace System.Net
 {
     internal static partial class NameResolutionPal
     {
         public const bool SupportsGetAddrInfoAsync = false;
 
+        [UnsupportedOSPlatformGuard("wasi")]
+        public static bool SupportsGetNameInfo = !OperatingSystem.IsWasi();
```

Comment:
I'm not 100% sure UnsupportedOSPlatformGuard works with static fields, other places use property:

```suggestion
        public static bool SupportsGetNameInfo => !OperatingSystem.IsWasi();
```